### ISSUE TITLE
Not configuration file given, using default file DDS_ROUTER_CONFIGURATION.yaml

### ISSUE THREAD

**Issue by magladko** [eProsima/DDS-Router#404]
Hi,
I downloaded the docker image of DDS-Router, but I get this messages:

```
kornel@kornel-LG:~/Programs/dds_router_ws$ docker run -it --net=host -v /home/kornel/Programs/dds_router_ws/DDS_ROUTER_CONFIGURATION.yaml:/root/DDS_ROUTER_CONFIGURATION.yaml ubuntu-ddsrouter:v2.0.0
Not configuration file given, using default file DDS_ROUTER_CONFIGURATION.yaml.
Starting DDS Router Tool execution.
DDS Router running.
```

I am specifically worried about 'Not configuration file given, using default file DDS_ROUTER_CONFIGURATION.yaml.' that I get despite the fact of creating a mounted drive with it present.

Source of my yaml file I copied from here:
https://eprosima-dds-router.readthedocs.io/en/latest/rst/examples/echo_example.html#examples-echo-example

Which is:
```
################
# ECHO EXAMPLE #
################

##################################
# CONFIGURATION VERSION
version: v3.0                                                         # 0

##################################
# ALLOWED TOPICS
# Allowing FastDDS and ROS2 HelloWorld demo examples topics

allowlist:
  - name: HelloWorldTopic                                             # 1
    type: HelloWorld                                                  # 1
  - name: rt/chatter                                                  # 2
    type: std_msgs::msg::dds_::String_

##################################
# PARTICIPANTS
participants:

##################################
# SIMPLE PARTICIPANT
# This participant will subscribe to topics in allowlist in specific domain and listen every message published there

  - name: SimpleParticipant                                           # 3
    kind: local                                                       # 4
    domain: 0                                                         # 5

##################################
# ECHO PARTICIPANT
# This Participant will print in stdout every message received by the other Participants, as well as discovery information

  - name: EchoParticipant                                             # 6
    kind: echo                                                        # 7
    discovery: true                                                   # 8
    data: true                                                        # 9
    verbose: true                                                     # 10


##################################
# CONFIGURATION DESCRIPTION

# This configuration example configures a DDS Router to listen to every message published in domain 0 in topics
# HelloWorldTopic (from Fast DDS HelloWorld) and rt/chatter from ROS2 demo_nodes, and to print the received
# messages in stdout. Information regarding discovery events is also printed to stdout.

# 0: Use YAML configuration version v3.0

# 1: Allow DDS Topic Name <HelloWorldTopic> with type <HelloWorld>.

# 2: Insert new topics in order to route them.

# 3: New Participant with name <SimpleParticipant>.

# 4: Kind of SimpleParticipant: <local>.
# LAN UDP communication with default simple multicast discovery.

# 5: SimpleParticipant will use DDS Domain ID <0>.

# 6: New Participant with name <EchoParticipant>.

# 7: Kind of EchoParticipant: <echo>.

# 8: Print a trace to stdout every time an Endpoint is discovered.

# 9: Print a trace to stdout every time a new data message arrives to the router.

# 10: Display verbose information regarding received messages (receiver endpoint_guid and data payload).
```

Then when I run ros2 example nodes:
export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
ros2 run demo_nodes_cpp listener / talker

I don't get any logs regarding communication between them.

**Comment by juanlofer-eprosima** [comment]
Hi @magladko,

I believe there is nothing wrong with that trace being printed out. It just means ´--config-file´ is not being provided, thus it attempts to load `DDS_ROUTER_CONFIGURATION.yaml` (which succeeds as otherwise the process would terminate with an error trace).

To double check it is well loaded, you may also try the equivalent command:
`docker run -it --net=host -v /home/kornel/Programs/dds_router_ws/DDS_ROUTER_CONFIGURATION.yaml:/root/my_config.yaml ubuntu-ddsrouter:v2.0.0 -c /root/my_config.yaml`.

Regarding the no communication between the talker/listener and the router, can you at least observe discovery events in the router or nothing at all? Also, may I ask how are you launching demo nodes? It may be that your talker/listener is in a docker network while your router is not (`--net=host`).

**Comment by magladko** [comment]
Thank you for your response @juanlofer-eprosima.
1. You are right, it seems that it is still using the provided yaml file. Just a quick note here, I don't know any way to pass arguments to router ran on docker, other than help and version, here is a print:
```
docker run -it --net=host -v /home/kornel/Programs/dds_router_ws/DDS_ROUTER_CONFIGURATION.yaml:/root/DDS_ROUTER_CONFIGURATION.yaml ubuntu-ddsrouter:v2.0.0 -h
-------------------------------------------------------------------
Entrypoint script for DDS Router Docker image
  -h|--help|help          Shows this help message
  -v|--version|version    Shows the repositories and versions of all the
                          installed tools and libraries

An empty argument list will launch the application
Any other argument will be executed on the image
-------------------------------------------------------------------
```
2. My nodes are run outside of the docker image, just plainly on my machine in another terminal. And yes, there is no information about any traffic.
This is all I get even after launching the nodes:
```
~$ docker run -it --net=host -v /home/kornel/Programs/dds_router_ws/DDS_ROUTER_CONFIGURATION.yaml:/root/DDS_ROUTER_CONFIGURATION.yaml ubuntu-ddsrouter:v2.0.0
Not configuration file given, using default file DDS_ROUTER_CONFIGURATION.yaml.
Starting DDS Router Tool execution.
DDS Router running.

```

**Comment by juanlofer-eprosima** [comment]
Hi @magladko,

I tried to recreate your setup and in my case everything works as expected. Regarding your questions:

1. You need to add the name of the executable (`ddsrouter`), note that I edited my previous comment as it was wrong.

2. This clearly should not happen. There might be a transport-related issue (summary: trying to use shared memory transport when that's not possible), however if this was the case you should at least see one or two discovery traces. Just in case, try addding `transport: udp` under `domain: 0` (at the same depth level), that should solve the transport issue.
Nonetheless, what version of ROS 2 are you using?

**Comment by magladko** [comment]
Hi @juanlofer-eprosima,
in fact, it works with the distribution you provided me.

I use ROS2 Humble directly installed on my system (not in docker). Everything is up to date and I use ROS2 built-in fastdds version (did not install it externally).

I tried adding the `transport: udp`, but withot luck. I also ensured that I correctly export `export RMW_IMPLEMENTATION=rmw_fastrtps_cpp` and `export ROS_DOMAIN_ID=0`

Not a docker expert, tried also adding ipc argument, but also did not work. 

Also just a note, I did not experience any errors while using my installation of ros2, so I am not sure where could I look next.

Still, thanks for your help.
